dom/index.js

```diff
@@ -42,8 +42,11 @@ function kebab (str) {
 
 function isInViewPort (el) {
   if (el && el.parentElement) {
-    const { top, bottom } = el.getBoundingClientRect()
-    return top < window.innerHeight && bottom >= 0
+    const { top, bottom, height } = el.getBoundingClientRect()
+    const isAboveTop = height > 0 && top + height <= 0
+    return isAboveTop
+      ? false
+      : top < window.innerHeight && bottom >= 0
```

Comment:
This is quite confusing. It looks like we're doing one thing if the element has height and another if it does not have height. Surely whatever we do should be consistent. Also why is `top + height` needed, surely that's the same as bottom?

_________ 0

___ top

___.bottom


_________ window.innerHeight

Reply:
I agree, it is confusing. Shall we pair on it a bit tomorrow and talk it through? Its hard to type it all out and have it make sense...